### PR TITLE
Include all primary authors in MARCXML export

### PR DESCRIPTION
**Pull request by nop33**
Fixes ticket `RQF0683434`

### CODE REVIEW

File: indico/MaKaC/common/output.py [hunk 1]
```diff
@@ -494,40 +494,30 @@ def _generate_contrib_location_and_time(self, contrib, out):
...
-                    list_[user].append('Author')
+        users = {}
+        primary_authors = list(contrib.primary_authors)
+        secondary_authors = list(contrib.secondary_authors)
+        for primary_author in primary_authors:
+            users[primary_author] = ['Primary Author']
+
+        for user in secondary_authors:
+            if user in users:
+                users[user].append('Author')
```
Comment (ThiefMaster) [inline indico/MaKaC/common/output.py]: why not make users a `defaultdict(list)`?

File: indico/MaKaC/common/output.py [hunk 2]
```diff
@@ -494,40 +494,30 @@ def _generate_contrib_location_and_time(self, contrib, out):
...
-            out.writeTag('subfield', u'{} {}'.format(auth.last_name, auth.first_name), [['code', 'a']])
-            for val in list_[user]:
-                out.writeTag('subfield', val, [['code', 'e']])
-            out.writeTag('subfield', auth.affiliation, [['code', 'u']])
-            out.closeTag('datafield')
-            del list_[auth]
-
-        for user in list_.keys():
-            out.openTag('datafield', [['tag', '700'], ['ind1', ' '], ['ind2', ' ']])
+        for user in users.keys():
```
Comment (ThiefMaster) [inline indico/MaKaC/common/output.py]: you get the same by iterating over `users` without `.keys()` - also, I'd use iteritems instead of accessing `users[user]` later

File: indico/MaKaC/common/output.py [hunk 3]
```diff
@@ -494,41 +495,25 @@ def _generate_contrib_location_and_time(self, contrib, out):
...
-        if auth:
-            auth = auth[0]
-            list_[auth] = ['Primary Author']
-        for user in aList:
-            if user in list_:
-                if user != auth:
-                    list_[user].append('Author')
-            else:
-                list_[user] = ['Author']
+        users = defaultdict(list)
```
Comment (OmeGak) [inline indico/MaKaC/common/output.py]: I would use `persons` for clarity. We are dealing with `User` not with `PersonLink` here.

Comment (nop33) [inline indico/MaKaC/common/output.py]: Well, I won't bother changing it now, it's code that will be rewritten.

File: indico/MaKaC/common/output.py [hunk 4]
```diff
@@ -494,41 +495,25 @@ def _generate_contrib_location_and_time(self, contrib, out):
...
+        primary_authors = list(contrib.primary_authors)
```
Comment (ThiefMaster) [inline indico/MaKaC/common/output.py]: why is that needed? you never modify it

File: indico/MaKaC/common/output.py [hunk 5]
```diff
@@ -494,41 +495,25 @@ def _generate_contrib_location_and_time(self, contrib, out):
...
+        secondary_authors = list(contrib.secondary_authors)
```
Comment (ThiefMaster) [inline indico/MaKaC/common/output.py]: same here
